beard/metrics/tests/test_text.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Beard.
+# Copyright (C) 2015 CERN.
+#
+# Beard is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""Test text metrics.
+
+.. codeauthor:: Petros Ioannidis <petros.ioannidis91@gmail.com>
+
+"""
+from __future__ import generators
+import pytest
+from pytest import mark
+from ..text import find_all, _jaro_matching, jaro, jaro_winkler
+
+
+@mark.parametrize('s, letter, occur',
+                  (('MARTHA', 'A', (1, 5)),
+                   ('DWAYNE', 'D', (0, )),
+                   ('A', 'A', (0, )),
+                   ('AABAA', 'AA', (0, 3)),
+                   ('ABCD', 'D', (3, ))))
+def test_find_all_normal_string(s, letter, occur):
+    """Test find_all behaviour for average cases."""
+    assert tuple(find_all(s, letter)) == occur
+
+
+@mark.parametrize('s, letter',
+                  (('MARTHA', 'Z'),
+                   ('', 'A')))
+def test_find_all_none_string(s, letter):
+    """Test find_all behaviour for empty cases."""
+    with pytest.raises(StopIteration):
+        assert next(find_all(s, letter))
+
+
+@mark.parametrize('s, letter',
+                  ((set(), 'A'),
+                   (dict(), 'A'),
+                   (int(), 'A'),
+                   (float(), 'A'),
+                   (list(), 'A')))
+def test_find_all_abnormal_string(s, letter):
+    """Test find_all behaviour called with wrong objects."""
+    with pytest.raises(TypeError):
+        next(find_all(s, letter))
+
+
+@mark.parametrize('s1, s2, match',
+                  (('MARTHA', 'MARHTA', (6, 2)),
+                   ('DWAYNE', 'DUANE', (4, 0)),
+                   ('DUANE', 'DWAYNE', (4, 0)),
+                   ('MARHTA', 'MARTHA', (6, 2))))
+def test_jaro_matching(s1, s2, match):
+    """Test jaro_matching behaviour."""
+    assert _jaro_matching(s1, s2) == match
+
+
+@mark.parametrize('s1, s2, match',
+                  (('MARTHA', 'MARHTA', 0.944),
+                   ('DWAYNE', 'DUANE', 0.822)))
+def test_jaro(s1, s2, match):
```

Comment:
Can you test the border cases as well? ie. when strings are identical / dont share any letters / one is empty.